contracts/pool/src/tests/swap.rs

```diff
@@ -639,3 +639,76 @@ fn swap_simulation_one_third_pool() {
         }
     );
 }
+
+#[test]
+fn bug_issue_169() {
+    let env = Env::default();
+    env.mock_all_auths();
+    env.budget().reset_unlimited();
+
+    let mut token1 = deploy_token_contract(&env, &Address::random(&env));
+    let mut token2 = deploy_token_contract(&env, &Address::random(&env));
+    if token2.address < token1.address {
+        std::mem::swap(&mut token1, &mut token2);
+    }
+
+    let swap_fees = 1_000i64; // 10% bps
```

Comment:
Can you try smaller amounts, 1% and 0.3%?

Reply:
got the expected results